### PR TITLE
Add custom field to Node Record

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Adds a custom field to Node Record object and API to update it through Discovery System.

### CODE REVIEW

File: src/main/java/org/ethereum/beacon/discovery/schema/IdentitySchemaV4Interpreter.java [hunk 1]
```diff
@@ -116,4 +128,15 @@ public NodeRecord createWithNewAddress(
   private static InetAddress getInetAddress(final Bytes address) throws UnknownHostException {
     return InetAddress.getByAddress(address.toArrayUnsafe());
   }
+
+  private static Stream<EnrField> getAllFields(final NodeRecord nodeRecord) {
```
Comment (ajsutton) [inline src/main/java/org/ethereum/beacon/discovery/schema/IdentitySchemaV4Interpreter.java:132]: nit: `streamAllFields` tends to be the convention for things that return a `Stream`.

File: src/main/java/org/ethereum/beacon/discovery/schema/NodeRecordBuilder.java [hunk 2]
```diff
@@ -70,6 +75,11 @@ static void addFieldsForUdpAddress(
     fields.add(new EnrField(isIpV6 ? EnrField.UDP_V6 : EnrField.UDP, udpPort));
   }
 
+  static void addCustomField(
+      final List<EnrField> fields, final String fieldName, final Bytes value) {
+    fields.add(new EnrField(fieldName, value));
+  }
```
Comment (ajsutton) [inline src/main/java/org/ethereum/beacon/discovery/schema/NodeRecordBuilder.java:81]: Does this really need to be it's own method?

Comment (cemozerr) [reply src/main/java/org/ethereum/beacon/discovery/schema/NodeRecordBuilder.java:81]: Not really but I wanted to make sure I adhere to the convention.

File: src/main/java/org/ethereum/beacon/discovery/schema/IdentitySchemaV4Interpreter.java (hunk 1 above)
Comment (cemozerr) [reply src/main/java/org/ethereum/beacon/discovery/schema/IdentitySchemaV4Interpreter.java:132]: done.

**Pull request merged**